base/intfuncs.jl

```diff
@@ -755,9 +759,17 @@ function digits!(a::AbstractVector{T}, n::Integer; base::Integer = 10) where T<:
     isempty(a) && return a
 
     if base > 0
-        for i in eachindex(a)
-            n, d = divrem(n, base)
-            a[i] = d
+        if ispow2(base) && n >= 0 && n isa Base.BitInteger
+            k = trailing_zeros(base)
+            c = base - 1
+            for i in eachindex(a)
+                a[i] = (n >> (k * (i - firstindex(a)))) & c
```

Comment:
For the record: here it's assumed that `firstindex(x)` is an integer and that the indices of `a` are consecutive, which is not enforced by the `AbstractArray` interface apparently. The same assumption is made below in the `base < 0` branch, so anyway this is IMHO not blocking merging this PR.